src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -227,6 +263,14 @@ CHIP_ERROR OnTransferSessionBegin(TransferSession::OutputEvent & event)
             mSystemLayer->CancelTimer(HandleBdxInitReceivedTimeoutExpired, this);
         }
 
+        // Start a timer for the BDX transfer timeout
```

Comment:
Why do we need a timer here, apart from the existing exchange response timer?  Maybe I'm just misunderstanding the setup, but it seems like the exchange timer exactly implements the "we expected a response by some time, but it did not come" behavior....

Reply:
With the poll output code the bdx timeout timer was in the transfer session itself. Last I spoke to Carol the idea was for the requestor and provider to track if BDX has stalled for more than 5 minutes and if it is we abort the transfer. The change was made for the requestor as part of the clean up but the provider was not updated. We discussed and made this change accordingly.

Reply:
The time-out should be at the exchange layer if at all possible (see src/messaging/ExchangeContext.h):

```
// Set the response timeout for the exchange context, regardless of the underlying session type. Using
    // UseSuggestedResponseTimeout to set a timeout based on the type of the session and the application processing time instead of
    // using this function is recommended.
    //
    // If a timeout of 0 is provided, it implies no response is expected. Consequently, ExchangeDelegate::OnResponseTimeout will not
    // be called.
    //
    void SetResponseTimeout(Timeout timeout);
```